### DISCUSSION TITLE
Enhancements

### DISCUSSION THREAD

**Discussion by Vilo176** [canton7/PropertyChanged.SourceGenerator#35] [Ideas] [answered]
Hi Antony,

Your SG is a very nice piece of code, thanks a lot for sharing.
I have 2 "demands" for you :

1. Make Automatic DependsOn optional.
I have some cases that I want to manage manually.

2. When a property is derived from an interface, your naming system takes the fully qualified name.
```
namespace Namespace_Intf
{
	public interface iObj
	{
		bool Active{ get; }
	}
}

public class cObj : Namespace_Intf.iObj
{
	public Namespace_Intf.iObj Other{ get; set; }

	[PropertyChanged.SourceGenerator.DependsOn( nameof(Other))]
	bool Namespace_Intf.iObj.Active => Other.Active;
}
```
code generated (simplified) :
```
public override void OnPropertyChanged(PropertyChangingEventArgs eventArgs)
{
	...
	if (text == "Other")
	{
		OnPropertyChanged(new PropertyChangedEventArgs("Namespace_Intf.iObj.Active"));
	}
	...
}
```
But this brokes many mechanisms (WPF Binding, ...).
You could make that optional too (ie. ```Namespace_Intf.iObj.Active``` => ```Active```).

Thanks 👍

**Comment by canton7** [comment]
> Make Automatic DependsOn optional. I have some cases that I want to manage manually.

That's fair enough. Would a .editorconfig setting at the project level work? Can you open an issue?

Explicit interface implementations is an interesting one. Binding to explicitly-implemented properties is badly supported, see https://github.com/dotnet/wpf/issues/3855. I also can't find any guidance on what PropertyChanged events should be raised for explicitly implemented properties. For example:

```cs
public class Test : ITest
{
    string ITest.Name => "Foo";
    public string Name => "Bar";
}
```

If you raise a PropertyChanged event with a propertyName of "Name", which property should be accessed? I'd say obviously the one returning "Bar". Which means that raising an event with a propertyName of "Name" when `ITest.Name` changes is wrong.

**Reply by Vilo176** [reply]
WPF Binding with explicit interface works as far as the Binding itself : the Binding mechanism is able to find the right property inside the object (```ITest.Name``` in your sample code).

But for _not obvious_ reasons, developers have not push the logic to its end, because when you raise ```PropertyChanged``` with the FQN, it doesn't work.
Consequently, user must raise with the simple name, hence my demand.

This is not logic/clean, but this is WPF bad.

**Reply by canton7** [reply]
Fixed in 46c05794d69e38d7bd4220e1936f28c47af4ff71

**Comment by canton7** [comment, marked as answer]
Both released in 1.0.7
